cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -212,38 +378,91 @@ public void testBuildCollectionNameDMByEntityRootServicePath() {
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - There was some problem when building the collection name");
+            assertTrue(false);
+        } // catch
+    } // testBuildCollectionNameDMByEntityRootServicePathOldEncoding
     
     /**
      * [NGSIMongoBaseSink.buildCollectionName] -------- When / service-path is notified/defaulted and
-     * data_model=dm-by-attribute, the MongoDB collections name is
-     * <prefix>/_<entityId>_<entityType>_<attrName>_<attrType>.
+     * data_model=dm-by-attribute, the MongoDB collections name is the concatenation of \<prefix\>, \<service-path\>,
+     * \<entityId\>, \<entityType\> and \<attrName\>.
      */
     @Test
-    public void testBuildCollectionNameDMByAttributeRootServicePath() {
+    public void testBuildCollectionNameDMByAttributeRootServicePathNewEncoding() {
         System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                 + "-------- When / service-path is notified/defaulted and data_model=dm-by-attribute, the "
-                + "MongoDB collections name is <prefix>/_<entityId>_<entityType>_<attrName>_<attrType>");
+                + "MongoDB collections name is the concatenation of <prefix>, <service-path>, <entityId>, <entityType> "
+                + "and <attrName>");
         String collectionPrefix = "sth_";
         String dbPrefix = "sth_";
         String dataModel = "dm-by-attribute";
+        String enableEncoding = "true";
         NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
-        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel));
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
         String dbName = "sth_default";
         String fiwareService = "default";
         String fiwareServicePath = "/";
-        String entity = "someId_someType";
-        String attribute = "someName_someType";
+        String entity = "someId=someType";
+        String attribute = "someName";
         boolean isAggregated = false;
         String entityId = "someId";
         String entityType = "someType";
         
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002fxffffsomeIdxffffsomeTypexffffsomeName";
             
             try {
-                assertEquals("sth_/_someId_someType_someName_someType", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4